### PR TITLE
login unit test, refactored app.py

### PR DESCRIPTION
**Pull request by clarawong20**
created unit testing for the login function. checks for the following:

- login page is accessible
- login with valid user credentials
- login with invalid credentials

also refactored app.py so that it would create and configure the app for testing purposes

### CODE REVIEW

File: app/src/app.py [hunk 1]
```diff
@@ -1,14 +1,26 @@
 from flask import Flask, render_template
 from models import db, User
```
Comment (zareentk) [inline app/src/app.py]: There seems to be two db's, one could potentially be removed

File: tests/test_login.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        # send HTTP POST request to the /login route with test user info, simulates the process of user submitting a login form
+        response = self.client.post('/login', data={'email': 'test@example.com', 'password': 'testpassword'}) 
+        # redirect on successful login, 302 = "found" redirection response
+        self.assertEqual(response.status_code, 302)
+
+    def TestLoginInavlidUser(self):
+        # send HTTP POST request with wrong user info
+        response = self.client.post('/login', data={'email': 'nonexistent@example.com', 'password': 'wrongpassword'})
+        # re-render login page, 200 means 'ok' as well as unsuccessful login attempt
+        self.assertEqual(response.status_code, 200)
```
Comment (zareentk) [inline tests/test_login.py:58]: A few more assertions could be helpful in testing

**Review by zareentk — CHANGES_REQUESTED** [review]

File: app/src/app.py (hunk 1 above)
Comment (skydonline) [inline app/src/app.py]: Relative imports, from local files from within the project, need to have a `.` infront of them. So it should be `.models`. External packages, like `flask`, do not need to have a `.` infront of them.

**Review by skydonline — CHANGES_REQUESTED** [review]
Please update the import, check the comment. Only need to update to `.models`

**Review by skydonline — APPROVED** [review]

**Review by skydonline — APPROVED** [review]

**Review by zareentk — APPROVED** [review]
Looks great!

**Pull request merged**